client/blocks/qr-code-login/index.jsx

```diff
@@ -0,0 +1,198 @@
+import { getTracksAnonymousUserId } from '@automattic/calypso-analytics';
+import { Card } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import QRCode from 'qrcode.react';
+import { useEffect, useState, useRef } from 'react';
+import qrCenter from 'calypso/assets/images/qr-login/wp.png';
+
+import './style.scss';
+
+const AUTH_PULL_INTERVAL = 5000; // 5 seconds
+const LOCALE_STORAGE_KEY = 'qr-login-token';
+
+const isStillValidToken = ( tokenData ) => {
+	if ( ! tokenData ) {
+		return false;
+	}
+	const { expires } = tokenData;
+	if ( ! expires ) {
+		return false;
+	}
```

Comment:
Seems like this could be shortened:

```suggestion
	if ( ! tokenData?.expires ) {
		return false;
	}
```